### PR TITLE
Add AI Block validation for AI Dance Party

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR pairs with https://github.com/code-dot-org/dance-party/pull/656 from `dance-party`.
These PRs add validation support required by the new AI block used in the HOC Dance Party AI activity progression.

I had issues with merging with staging's version of `yarn.lock` so closed [this PR](https://github.com/code-dot-org/code-dot-org/pull/54145), and opened this one with [same updates](https://github.com/code-dot-org/code-dot-org/pull/54145/files#diff-2732dd862d61c871c77540f498002e18cc0837568ec0c10fcad8d32b437d0076) to `Dance.js`. This PR also upgrades the version of `@code-dot-org/dance-party` to v1.0.9.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/dance/Dance.js [hunk 1]
```diff
@@ -628,12 +628,22 @@ Dance.prototype.execute = async function () {
...
-    this.nativeAPI.play(songMetadata, success => {
-      this.performanceData_.lastRunButtonDelay =
-        performance.now() - this.performanceData_.lastRunButtonClick;
-      success ? resolve() : reject();
+  const userBlockTypes = [];
+  Blockly.getMainWorkspace()
+    .getAllBlocks()
+    .forEach(block => {
+      userBlockTypes.push(block.type);
     });
```
Comment (breville) [inline apps/src/dance/Dance.js:636]: nit: you could use `.map` here?

**Review by breville — APPROVED** [review]
This seems alright, but before merging, please double-check that the published version works correctly in your local repo, and that all of the new validation functionality works end-to-end as expected.

**Comment by fisher-alice** [comment]
> This seems alright, but before merging, please double-check that the published version works correctly in your local repo, and that all of the new validation functionality works end-to-end as expected.

Thanks! Just checked locally with new published version and works as expected.

File: apps/src/dance/Dance.js (hunk 1 above)
Comment (fisher-alice) [reply apps/src/dance/Dance.js:636]: I'm sorry - I didn't see this till after I merged. I'll add this change to my next PR https://github.com/code-dot-org/code-dot-org/pull/54218

**Pull request merged**